lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -4490,37 +4526,42 @@ LogicalResult FIRRTLLowering::lowerVerificationStatement(
         assumeLabel = StringAttr::get(builder.getContext(),
                                       "assume__" + label.getValue());
       addToIfDefBlock("USE_PROPERTY_AS_CONSTRAINT", [&]() {
-        builder.create<sv::AssumeConcurrentOp>(
-            circt::sv::EventControlAttr::get(builder.getContext(), event),
-            clock, predicate, assumeLabel);
+        if (!isUnrOnlyAssert) {
+          builder.create<sv::AssumeConcurrentOp>(
+              circt::sv::EventControlAttr::get(builder.getContext(), event),
+              clock, predicate, assumeLabel);
+        } else {
+          builder.create<sv::AlwaysOp>(
+              ArrayRef(sv::EventControl::AtEdge), ArrayRef(predicate), [&]() {
+                buildImmediateVerifOp(builder, "assume", predicate,
+                                      circt::sv::DeferAssertAttr::get(
+                                          builder.getContext(),
+                                          circt::sv::DeferAssert::Immediate),
```

Comment:
@seldridge  looking at this code more I am pretty sure there is a "cleaner" way to get the whole true Immediate Op  (i.e. without clock) rather than what we have - not sure what it entails to get a version of `AssumeImmediateOp` without the `posedge clock` (which is what  seems to be happening now - hence had to explicitly call out the `AlwaysOp` here)